include/libtorrent/block_cache.hpp

```diff
@@ -337,7 +332,7 @@ namespace libtorrent
 		struct hash_value
 		{
 			std::size_t operator()(cached_piece_entry const& p) const
-			{ return std::size_t(p.storage.get()) + std::size_t(p.piece); }
+			{ return std::size_t(p.storage.get()) + static_cast<int>(p.piece); }
```

Comment:
it won't actually cost anything though. ``p.piece`` is already an ``int``. it would presumably already be 64 bits wide in the register, or sign-extended when loaded from memory.